modules/core/include/opencv2/core/types.hpp

```diff
@@ -63,6 +63,176 @@ namespace cv
+
+#define CV_ENUM_FLAGS(EnumType)                                                                       \
+__CV_ENUM_FLAGS_LOGICAL_NOT      (EnumType);                                                          \
+__CV_ENUM_FLAGS_LOGICAL_EQ       (int,      EnumType);                                                \
+__CV_ENUM_FLAGS_LOGICAL_NOT_EQ   (int,      EnumType);                                                \
+__CV_ENUM_FLAGS_LOGICAL_EQ       (EnumType, int);                                                     \
+__CV_ENUM_FLAGS_LOGICAL_NOT_EQ   (EnumType, int);                                                     \
+                                                                                                      \
+__CV_ENUM_FLAGS_BITWISE_NOT      (EnumType);                                                          \
+__CV_ENUM_FLAGS_BITWISE_OR       (EnumType, EnumType, EnumType);                                      \
+__CV_ENUM_FLAGS_BITWISE_AND      (EnumType, EnumType, EnumType);                                      \
+__CV_ENUM_FLAGS_BITWISE_XOR      (EnumType, EnumType, EnumType);                                      \
+                                                                                                      \
+__CV_ENUM_FLAGS_BITWISE_OR       (EnumType, int,      EnumType);                                      \
+__CV_ENUM_FLAGS_BITWISE_AND      (EnumType, int,      EnumType);                                      \
+__CV_ENUM_FLAGS_BITWISE_XOR      (EnumType, int,      EnumType);                                      \
+__CV_ENUM_FLAGS_BITWISE_OR       (EnumType, EnumType, int);                                           \
+__CV_ENUM_FLAGS_BITWISE_AND      (EnumType, EnumType, int);                                           \
+__CV_ENUM_FLAGS_BITWISE_XOR      (EnumType, EnumType, int);                                           \
+                                                                                                      \
+__CV_ENUM_FLAGS_BITWISE_OR_EQ    (EnumType, EnumType);                                                \
+__CV_ENUM_FLAGS_BITWISE_AND_EQ   (EnumType, EnumType);                                                \
+__CV_ENUM_FLAGS_BITWISE_XOR_EQ   (EnumType, EnumType);                                                \
+__CV_ENUM_FLAGS_BITWISE_OR_EQ    (EnumType, int);                                                     \
+__CV_ENUM_FLAGS_BITWISE_AND_EQ   (EnumType, int);                                                     \
+__CV_ENUM_FLAGS_BITWISE_XOR_EQ   (EnumType, int);                                                     \
+
+#define CV_ENUM_CLASS_FLAGS(EnumType)                                                                 \
+  CV_ENUM_FLAGS                  (EnumType);                                                          \
+__CV_ENUM_FLAGS_ARITHMETIC_PLUS  (int,      EnumType);                                                \
```

Comment:
We may want to get rid of those operators in the future, because they violate type-safety. If needed, users can always explicitly cast.

Reply:
@hrnr I introduced these temporarily for nasty things like `int flags = cv::RANSAC | ACCESS_WRITE;`, but I plan to get ride of it after I manage to convert several tangled-together enum types.

BTW: Thanks for the offer, I will let you know once I need a hand.

Reply:
great. I know this needs to go step by step, it is such a gargantuan task.